### ISSUE TITLE
Can I create one ComputeBuffer, repeatedly set data in it?

### ISSUE THREAD

**Issue by solaaa** [clSharp/Cloo#12]
Hi,

I met a problem that I wanted to repeatedly execute a kernel with different data:
for example, 100 times addition of two 3*3 matrice : matrix_A + matrix_B
at each time, I'd like to set different values of A and B.

now, all I do is 
1. new ComputeBuffer ( ..., matrix_A ), new ComputeBuffer ( ..., matrix_B )
2. execute kernel
3.  read and dispose the 2 buffer
4. repeat 1~3 steps

Can I just creat new ComputeBuffer() only once, and repeatedly set different value of the buffer to execute the kernel? when 100 times additions finish, I dispose it? 
I didn't fine a function to do this.
hope for all your response.

Thanks and Regards!
solaaa

**Comment by TillAlex** [comment]
Hi,

you can use the queue.Write(...) method to copy new data to a buffer.

Cheers,
Alex

P.S.: I guess adding two 3*3 matrices is just an example, but in case it's not: Be aware that you won't achieve any speedup by shifting the work to the GPU if the problem sizes are that small.

**Comment by solaaa** [comment]
> Hi,
> 
> you can use the queue.Write(...) method to copy new data to a buffer.
> 
> Cheers,
> Alex
> 
> P.S.: I guess adding two 3*3 matrices is just an example, but in case it's not: Be aware that you won't achieve any speedup by shifting the work to the GPU if the problem sizes are that small.

Hi, Alex,
Sincerely thanks for your reply!
With your help, I can correctly run my code.
Actually, I am a beginner of paralell programing and faced with many "tiny" problems, such as, to find my devices(CPU, GPU), to choose device by name, to use local threads, etc.
Is there any documentation about Cloo? 

Regards!
solaaa

**Comment by TillAlex** [comment]
You are welcome! Actually there is no documentation for Cloo. In most cases it's very close to the original OpenCL API.

Just as an example here is the documentation for the clEnqueueWriteBuffer method:
https://www.khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/clEnqueueWriteBuffer.html

It's signature is very similar to queue.Write(...). The differences come from the fact that Cloo is object oriented in contrast to the C-Api.

So basically you should read the OpenCL documentation and non Cloo based OpenCL tutorial. It should be easy to find the corresponding Cloo methods then.

**Comment by solaaa** [comment]
> You are welcome! Actually there is no documentation for Cloo. In most cases it's very close to the original OpenCL API.
> 
> Just as an example here is the documentation for the clEnqueueWriteBuffer method:
> https://www.khronos.org/registry/OpenCL/sdk/1.0/docs/man/xhtml/clEnqueueWriteBuffer.html
> 
> It's signature is very similar to queue.Write(...). The differences come from the fact that Cloo is object oriented in contrast to the C-Api.
> 
> So basically you should read the OpenCL documentation and non Cloo based OpenCL tutorial. It should be easy to find the corresponding Cloo methods then.

Thanks again!
I will learn and search the original OpenCL API firstly if meeting problems.
